packages/plugin-core/src/_extension.ts

```diff
@@ -1053,42 +1057,69 @@ async function _setupCommands(
     );
   }
 
-  if (!existingCommands.includes(DENDRON_COMMANDS.SHOW_PREVIEW.key)) {
-    context.subscriptions.push(
-      vscode.commands.registerCommand(
-        DENDRON_COMMANDS.SHOW_PREVIEW.key,
-        sentryReportingCallback(async () => {
-          await new ShowPreviewCommand(PreviewPanelFactory.create(ws)).run();
-        })
-      )
-    );
-  }
+  const addCommand = ({
+    context,
+    key,
+    cmd,
+    existingCommands,
+  }: {
+    context: vscode.ExtensionContext;
+    key: string;
+    cmd: IBaseCommand | (() => IBaseCommand);
+    existingCommands: string[];
+  }) => {
+    if (!existingCommands.includes(key)) {
+      context.subscriptions.push(
+        vscode.commands.registerCommand(
+          key,
+          sentryReportingCallback(async (args) => {
+            if (_.isFunction(cmd)) {
+              await cmd().run(args);
+            } else {
+              await cmd.run(args);
+            }
+          })
+        )
+      );
+    }
+  };
 
-  if (!existingCommands.includes(DENDRON_COMMANDS.SHOW_SCHEMA_GRAPH.key)) {
-    context.subscriptions.push(
-      vscode.commands.registerCommand(
-        DENDRON_COMMANDS.SHOW_SCHEMA_GRAPH.key,
-        sentryReportingCallback(async () => {
-          await new ShowSchemaGraphCommand(
-            SchemaGraphViewFactory.create(ws)
-          ).run();
-        })
-      )
-    );
-  }
+  addCommand({
+    context,
+    key: DENDRON_COMMANDS.SHOW_PREVIEW.key,
+    cmd: () => new ShowPreviewCommand(PreviewPanelFactory.create(ws)),
+    existingCommands,
+  });
 
-  if (!existingCommands.includes(DENDRON_COMMANDS.SHOW_NOTE_GRAPH.key)) {
-    context.subscriptions.push(
-      vscode.commands.registerCommand(
-        DENDRON_COMMANDS.SHOW_NOTE_GRAPH.key,
-        sentryReportingCallback(async () => {
-          await new ShowNoteGraphCommand(
-            NoteGraphPanelFactory.create(ws)
-          ).run();
-        })
-      )
-    );
-  }
+  // // NOTE: graph commands with create a panel when instantiated, hence passing them in via a closure
+  addCommand({
+    context,
+    key: DENDRON_COMMANDS.SHOW_SCHEMA_GRAPH.key,
+    cmd: () => new ShowSchemaGraphCommand(SchemaGraphViewFactory.create(ws)),
+    existingCommands,
+  });
+
+  addCommand({
+    context,
+    key: DENDRON_COMMANDS.SHOW_NOTE_GRAPH.key,
+    cmd: () => new ShowNoteGraphCommand(NoteGraphPanelFactory.create(ws)),
```

Comment:
If we are subscribing this closure instantiator function to a subscription event we are ok with new instances of the command being instantiated for each subscription event correct?

Reply:
good catch. reverted this part